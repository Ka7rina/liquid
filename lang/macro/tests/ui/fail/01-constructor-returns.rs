use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    impl Noop {
        #[liquid(constructor)]
        fn invalid_return(&mut self) -> Self {}

        #[liquid(message)]
        fn noop(&self, i: u8, s: String) -> (bool, i32) {}
    }
}

fn main() {}
