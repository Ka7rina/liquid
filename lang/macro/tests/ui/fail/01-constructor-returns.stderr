error: constructor of liquid contract should not have return value
  --> $DIR/01-constructor-returns.rs:10:38
   |
10 |         fn invalid_return(&mut self) -> Self {}
   |                                      ^^^^^^^
